src/shared/components/query/QueryStore.ts

```diff
@@ -1104,6 +1108,62 @@ export class QueryStore
     {
         return this.genesetIdsQuery.query;
     }
+    
+    readonly volcanoPlotTableData = remoteData<Geneset[]>({
+        invoke: async()=>{
+            const tableData: Geneset[] = [];
+            const hierarchyData = await getHierarchyData(

```

Comment:
this fetch should go in a util file.  also it should be decoupled from the logic which processes the result so that you can easily write a test of the processing logic.  by processing logic i mean the for loop that populates the table.